lib/compress.js

```diff
@@ -659,12 +665,18 @@ merge(Compressor.prototype, {
                             stat = stat.clone();
                             stat.condition = stat.condition.negate(compressor);
                             var body = as_statement_array(stat.alternative).concat(ret);
-                            var funs = extract_functions_from_statement_array(body);
```

Comment:
I think it's safe to say this part would be more readable and performant than https://github.com/mishoo/UglifyJS2/pull/1462/files#diff-230c53e13e3ee367aec7d49470eb42caL907